packages/core/src/state-management/repository.ts

```diff
@@ -943,15 +949,17 @@ export class Repository {
         .then((found) => {
           const state$ = found || new RunningState(init, false)
           this.inmemory.endure(id.toString(), state$)
-          state$.subscribe(subscriber).add(() => {
-            if (state$.observers.length === 0) {
+          const subscription = state$.subscribe(subscriber)
+          state$.add(subscription)
+          subscription.add(() => {
+            if (state$.subscriptionSet.size === 0) {
```

Comment:
Can you explain to me the significance of this change from checking `observers.length` to `subscriptionSet.size`?

Reply:
@JulissaDantes Additional comment to respond to.

Reply:
@stbrody @dbcfd The feed is going to be a subscriber to the state, and we dont want that to prevent the memory from being freed, so it is checking the `subscriptionSet.size` instead. 

Also, [according to the code](https://github.com/ReactiveX/rxjs/blob/fdfc8acf74acc314be7184fbfbc2089294c9253d/src/internal/Subject.ts#L19) `observers` is soon to be deprecated in an upcomming version.